apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/stu3/providers/PartDEventTransformer.java

```diff
@@ -257,15 +258,48 @@ private static ExplanationOfBenefit transformClaim(PartDEvent claimGroup) {
      * throw new InvalidRifValueException(
      * "Service Provider ID Qualifier Code is invalid: " +
      * claimGroup.serviceProviderIdQualiferCode);
+     *   Code	    Code value
+     *   01        National Provider Identifier (NPI)
+     *   06        Unique Physician Identification Number (UPIN)
+     *   07        National Council for Prescription Drug Programs (NCPDP) provider identifier
+     *   08        State license number
+     *   11        Federal tax number
+     *   99        Other
      */
 
+    String identiferSystem = null;
+
     if (!claimGroup.getServiceProviderId().isEmpty()) {
-      eob.setOrganization(
-          TransformerUtils.createIdentifierReference(
-              TransformerConstants.CODING_NPI_US, claimGroup.getServiceProviderId()));
-      eob.setFacility(
-          TransformerUtils.createIdentifierReference(
-              TransformerConstants.CODING_NPI_US, claimGroup.getServiceProviderId()));
+      switch (claimGroup.getServiceProviderIdQualiferCode()) {
+        case "01":
+          identiferSystem = TransformerConstants.CODING_NPI_US;
+          break;
+        case "06":
+          identiferSystem = TransformerConstants.CODING_UPIN;
+          break;
+        case "07":
+          identiferSystem = TransformerConstants.CODING_NCPDP;
+          break;
+        case "08":
+          identiferSystem = TransformerConstants.CODING_STATE_LICENSE_NUMBER;
+          break;
+        case "11":
+          identiferSystem = TransformerConstants.CODING_FEDERAL_TAX_NUMBER;
+          break;
+        case "99":
+          identiferSystem = TransformerConstants.CODING_OTHER;
+          break;
+      }
+
+      if (StringUtils.isNotBlank(identiferSystem) && StringUtils.isNotEmpty(identiferSystem)) {
```

Comment:
From reading the JavaDoc on these methods, I'm pretty sure the `isNotEmpty(...)` is redundant.
```suggestion
      if (StringUtils.isNotBlank(identiferSystem)) {
```

Reply:
changed